### DISCUSSION TITLE
Cross-Compiling Linux Hello World Application

### DISCUSSION THREAD

**Discussion by Future-Electronics-Design-Center** [polarfire-soc/polarfire-soc-documentation#124] [How to] [answered]
If I am using the Yocto build environment for my Linux distribution targeting the MPFS Icicle kit, how do I setup build environment for a Makefile-based Hello World application?

Thanks,
-Tim

**Comment by lhanlyu** [comment, marked as answer]
You can build you application on your target with the 'mpfs-dev-cli' image. For example look at the sample applications installed on your target icicle-kit: i.e. /opt/microchip/gpio
You will see the source files and make file.

**Reply by Future-Electronics-Design-Center** [reply]
Thanks for the suggestion lhanlyu, but unfortunately, your suggestion doesn't work for my case.  I am trying to compile the Linux PCIe driver for DG0756 and the required kernel includes are not present in that image so looking to cross-compile in the VM I use for development.  I think this is the more typical use case for a software developer anyway.

**Reply by hughbreslin** [reply]
DG0756 targets a Splash Kit or PolarFire Eval kit, are you trying to compile the kernel driver for use on an Icicle Kit with a Splash / Eval kit connected to it? I think @lhanlyu suggestion stands for a Hello World application but this is not a hello world application, the requirements in the demo guide are for a Cent OS host and not directly for PolarFire SoC. Just to confirm, [this](https://www.microsemi.com/document-portal/doc_download/136594-dg0756-polarfire-fpga-pcie-endpoint-ddr3-and-ddr4-memory-controller-data-plane-demo-guide) is the copy of the doc I'm looking at.

**Reply by Future-Electronics-Design-Center** [reply]
Hi Hugh,

Yes, that is the one.  I was hoping to connect the Icicle-kit and the Splash kit together, and run the Linux pcie_app code that is part of DG0756. I did try to compile the driver directly on the Icicle kit but got the output attached.  I have been able (with a little modification to the Makefile) built the driver for Ubuntu and have it running with the Splash kit now.

I was trying to keep my original request simple asking for a method to cross-compile (rather than compile directly on the target) a Hello World application under my Linux VM.  Thanks

[Build Error Icicle Kit.txt](https://github.com/polarfire-soc/polarfire-soc-documentation/files/9093302/Build.Error.Icicle.Kit.txt)

**Reply by lhanlyu** [reply]
Perfect. As you are using yocto, you could use a recipe and compile using oe_runmake.
After compile you could also install the application. As an example please look at the polarfire-soc-linux-examples.bb

**Reply by Future-Electronics-Design-Center** [reply]
Thanks Ihanlyu. I will look at the example recipes and oe_runmake.
